### PR TITLE
fix: add validation for non-support of prometheus extension for Windows

### PR DESCRIPTION
**Pull request by vyta**
**Reason for Change**:

This adds validation for non-support of the prometheus extension for Windows nodes

**Issue Fixed**:

Refers to #727 

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/vlabs/validate.go [hunk 1]
```diff
@@ -596,6 +596,14 @@ func (a *Properties) validateExtensions() error {
 		if len(agentPool.Extensions) != 0 && (len(agentPool.AvailabilityProfile) == 0 || agentPool.IsVirtualMachineScaleSets()) {
 			return errors.Errorf("Extensions are currently not supported with VirtualMachineScaleSets. Please specify \"availabilityProfile\": \"%s\"", AvailabilitySet)
 		}
+
+		if agentPool.OSType == Windows && len(agentPool.Extensions) != 0 {
```
Comment (CecileRobertMichon) [inline pkg/api/vlabs/validate.go]: thanks for the PR @vyta ! Could you please add a small unit test to test this new validation?

Comment (CecileRobertMichon) [reply pkg/api/vlabs/validate.go]: https://github.com/Azure/aks-engine/blob/master/pkg/api/vlabs/validate_test.go#L865

**Comment by vyta** [comment]
@CecileRobertMichon tests failed, but I'm not entirely sure why they failed/how to fix?

File: pkg/api/vlabs/validate.go (hunk 1 above)
Comment (vyta) [reply pkg/api/vlabs/validate.go]: Added--let me know if that works

**Comment by mboersma** [comment]
@vyta the specific unit test failure was this:
```
--- FAIL: TestProperties_ValidateInvalidExtensions (0.00s)
    validate_test.go:909: expected error message : prometheus-grafana-k8s extension is currently not supported for Windows agents to be thrown, but got Extensions are currently not supported with VirtualMachineScaleSets. Please specify "availabilityProfile": "AvailabilitySet"
```

**Comment by vyta** [comment]
@mboersma thanks! I am struggling to get my envs to cooperate, so I couldn't test locally. Please bare with me as I may continue to push changes that fail tests

**Comment by CecileRobertMichon** [comment]
lint is failing:
```
golangci-lint has version 1.16.0 built from 97ea1cb on 2019-03-31T19:48:46Z
Running inside container
pkg/api/vlabs/validate_test.go:894: File is not `goimports`-ed (goimports)
					Name:   "agentpool",
					VMSize: "Standard_D2_v2",
					Count:  1,
Makefile:138: recipe for target 'test-style' failed
```

**Pull request merged**